Packs/GoogleChronicleBackstory/Integrations/GoogleChronicleBackstory/GoogleChronicleBackstory.py

```diff
@@ -1390,55 +2327,49 @@ def domain_command(client_obj, domain_name: str):
 def fetch_incidents(client_obj, params: Dict[str, Any]):
     """
     fetches alerts or IoC domain matches and convert them into actionable incidents.
-    :param client_obj:
-    :param params:
+
+    :type client_obj: Client
+    :param client_obj: object of the client class
+
+    :type params: dict
+    :param params: configuration parameter of fetch incidents
     :return:
     """
     first_fetch_in_days = params.get('first_fetch', '3 days').lower()  # 3 days as default
     max_fetch = params.get('max_fetch', 10)  # default page size
-    filter_severity = params.get('incident_severity', 'ALL')  # All to get all type of severity
+    time_window = params.get('time_window', '15')
 
     # getting numeric value from string representation
     start_time, end_time = parse_date_range(first_fetch_in_days, date_format=DATE_FORMAT)
 
-    last_run = demisto.getLastRun()
-    if last_run and 'start_time' in last_run:
-        start_time = last_run.get('start_time', start_time)
-
     # backstory_alert_type will create actionable incidents based on input selection in configuration
-    backstory_alert_type = params.get('backstory_alert_type', 'ioc domain matches')
+    backstory_alert_type = params.get('backstory_alert_type', 'ioc domain matches').lower()
 
     incidents = []
-    if 'ioc domain matches' != backstory_alert_type.lower():
-        events = get_gcb_alerts(client_obj, start_time, end_time, max_fetch, filter_severity)
-
-        _, contexts = group_infos_by_alert_asset_name(events)
 
-        # Converts event alerts into  actionable incidents
-        for event in list(contexts.values()):
-            severity = SEVERITY_MAP.get(event['Severities'].lower(), 0)
-            incident = {
-                'name': '{} for {}'.format(event['AlertName'], event['Asset']),
-                'details': json.dumps(event),
-                'severity': severity,
-                'rawJSON': json.dumps(event)
-            }
-            incidents.append(incident)
+    if "assets with alerts" == backstory_alert_type:
+        incidents = fetch_incidents_asset_alerts(client_obj, params, start_time, end_time, time_window, max_fetch)
+    elif "user alerts" == backstory_alert_type:
+        incidents = fetch_incidents_user_alerts(client_obj, params, start_time, end_time, time_window, max_fetch)
+    elif 'detection alerts' == backstory_alert_type:
+        incidents = fetch_incidents_detection_alerts(client_obj, params, start_time, end_time, time_window, max_fetch)
```

Comment:
Could you please elaborate about the change that was made to the fetch logic?
It seems that the new function here uses the V2 API - are you sure it's accessible for everyone?
Moreover, I'm a bit worried about backwards compatibility here so I'd appreciate a detailed explanation

Reply:
> Could you please elaborate about the change that was made to the fetch logic?
> It seems that the new function here uses the V2 API - are you sure it's accessible for everyone?
> Moreover, I'm a bit worried about backwards compatibility here so I'd appreciate a detailed explanation

@guykeller 

- We have added a provision to fetch two more variants of alert types that the chronicle instance provides as part of the fetch-incidents functionality on demisto. Additionally, we have also added one configuration parameter in order to handle late-breaking events of chronicle, which was reported by one of our customers. We have shared a beta file for the same with them and haven't received any issues related to it so far.

- Yes, the new function for detection uses the V2 API and it should be accessible for the customers of Chronicle.

- Lastly, we have taken care that we do not remove the existing configuration parameters or aspects of the existing fetch logic that might break the backward compatibility, and so we have just added some configuration parameters or added options in some of the existing ones, in order to provision the new functionalities.